framework/src/task_manager/task_manager_core.c

```diff
@@ -447,6 +447,67 @@ static void taskmgr_termination_callback(void)
 	mq_unlink(TM_PUBLIC_MQ);
 }
 
+static int taskmgr_check_msg_mask(int msg_mask, int handle)
+{
+	if (msg_mask & TASK_MSG_MASK(handle)) {
+		return OK;
+	}
+	return ERROR;
+}
+
+static int taskmgr_get_handle_by_pid(int pid)
+{
+	int chk_idx;
+	for (chk_idx = 0; chk_idx < CONFIG_TASK_MANAGER_MAX_TASKS; chk_idx++) {
+		if (TASK_PID(chk_idx) == pid) {
+			return chk_idx;
+		}
+	}
+	return ERROR;
+}
+
+static void taskmgr_broadcast(int msg)
+{
+	int chk_idx;
```

Comment:
chk_idx is not appropriate. task_idx is better I think because in your usages it means task..